### PR TITLE
Use auto-hiding scrollbars for codeview

### PR DESCRIPTION
**Pull request by billyeatcookies**
## Changes
- Codeview uses auto-hiding scrollbars instead of normal ones

### CODE REVIEW

**Comment by Moosems** [comment]
Honestly I really dislike auto-hiding scrollbars in general and it also complicated some app logic that uses winfo_reqwidth and winfo_reqheight.

**Comment by Moosems** [comment]
Maybe an option for using auto-hiding scroll bars with it set as False by default? This would also require some documentation and probably doesn't require a whole file.

**Comment by billyeatcookies** [comment]
@Moosems requested changes are made, please review again!

File: chlorophyll/codeview.py [hunk 1]
```diff
@@ -21,6 +21,20 @@
 LexerType = Union[Type[pygments.lexer.Lexer], pygments.lexer.Lexer]
 
 
+class Scrollbar(ttk.Scrollbar):
+    def __init__(self, master, autohide, *args, **kwargs):
+        super().__init__(master, *args, **kwargs)
+        self.autohide = autohide
+
+    def set(self, low, high):
+        if self.autohide:
```
Comment (Moosems) [inline chlorophyll/codeview.py:30]: Needs to have a grid() in an else for the main if statement.

**Review by Moosems — CHANGES_REQUESTED** [review]
Few changes/clarification necessary but perfect otherwise!

File: chlorophyll/codeview.py (hunk 1 above)
Comment (rdbende) [reply chlorophyll/codeview.py:30]: Actually, why?

File: chlorophyll/codeview.py [hunk 2]
```diff
@@ -21,6 +21,20 @@
...
+            if float(low) <= 0.0 and float(high) >= 1.0:
+                self.grid_remove()
+            else:
+                self.grid()
```
[the remark this answers is not in the document]
Comment (rdbende) [reply chlorophyll/codeview.py:34]: Yes. `grid_remove` saves the previous position, and `grid` will restore that.

**Review by rdbende — APPROVED** [review]

File: chlorophyll/codeview.py (hunk 1 above)
Comment (Moosems) [reply chlorophyll/codeview.py:30]: My bad, :).

**Review by Moosems — APPROVED** [review]
Looks good! Great job @billyeatcookies!

**Comment by Moosems** [comment]
@rdbende You want to merge?

**Comment by rdbende** [comment]
Whatever :) You have the permission too.

**Pull request merged**